### PR TITLE
Istio JWT full support for namespaces

### PR DESCRIPTION
**Pull request by videlov**
**Description**

Changes proposed in this pull request:

- Support correct namespace handling on all APIRule levels

**Related issue(s)**

https://github.com/kyma-project/api-gateway/issues/168

### CODE REVIEW

File: internal/processing/processors/request_authentication_processor.go [hunk 1]
```diff
@@ -92,5 +92,16 @@ func GetRequestAuthenticationKey(ra *securityv1beta1.RequestAuthentication) stri
...
 		jwtRulesKey += fmt.Sprintf("%s:%s", k.Issuer, k.JwksUri)
 	}
-	return fmt.Sprintf("%s:%s", ra.Spec.Selector.MatchLabels[RequestAuthenticationAppSelectorLabel], jwtRulesKey)
+
+	namespace := ra.Namespace
+	if namespace == "" {
+		namespace = "default"
+	}
+
+	return fmt.Sprintf("%s:%s:%s",
```
[the remark this answers is not in the document]
Comment (videlov) [reply internal/processing/processors/request_authentication_processor.go:101]: I raised this concern already on slack: https://sap-btp.slack.com/archives/C01L56SQ0QH/p1674824781074669

File: internal/validation/validate.go [hunk 2]
```diff
@@ -144,7 +144,7 @@ func (v *APIRule) validateService(attributePath string, api *gatewayv1beta1.APIR
 	for namespace, services := range v.ServiceBlockList {
 		for _, svc := range services {
 			serviceNamespace := helpers.FindServiceNamespace(api, nil)
-			if svc == *api.Spec.Service.Name && serviceNamespace != nil && namespace == *serviceNamespace {
+			if svc == *api.Spec.Service.Name && namespace == serviceNamespace {
```
Comment (barchw) [inline internal/validation/validate.go:147]: A nil check here may also make sense

**Review by barchw — CHANGES_REQUESTED** [review]
Looks good, just some small comments

**Review by barchw — APPROVED** [review]
/meow

**Pull request merged**